lib/src/api_analysis/README.md

```diff
@@ -0,0 +1,85 @@
+# API Analysis
+
+Refer to the [introductory blog post](https://arseny.uk/posts/gsoc2022/) before reading this document.
+
+## Running API analysis
+
+Three CLI subcommands are provided. Run `dart pub global activate --source git https://github.com/dart-lang/pana` to make them globally available and run them with `dart pub global run pana:api_analysis <subcommand>`.
+
+### `summary`
+
+```
+Displays a summary of the public API of a package.
+
+Usage: api-analysis summary <package-path>
+```
+
+This command writes a JSON *summary* of the package at `<package-path>`, which must point to the root directory of a package, to standard output.
+
+This package does not have to be published to pub.dev , and `<package-path>` can point to a directory in the [pub cache](https://dart.dev/tools/pub/glossary#system-cache).
+
+### `lower-bounds`
+
+```
+Runs lower bound constraint analysis on a single package.
+
+Usage: api-analysis lower-bounds <target-name>
+-c, --cache=<path>    Cache directory for requests to https://pub.dev/api/packages/PACKAGE-NAME .
+```
+
+This command performs *lower bound constraint analysis* on the latest version of the package named `<target-name>`, which must be the name of a package published to pub.dev . If at least one *issue* is discovered, a report is written to standard output. Any warnings are written to standard error.
+
+If `--cache=<path>` is not provided, a temporary folder is created automatically in the system temp directory and deleted after *analysis* exits.
+
+### `lower-bounds-batch`
+
+```
+Runs lower bound constraint analysis on many packages.
+
+Usage: api-analysis lower-bounds-batch <package-number> <process-number> <log-path>
```

Comment:
```suggestion
Usage: api-analysis lower-bounds-batch <max-packages> <concurrency> <log-path>
```

Reply:
doesn't `max-packages` suggest that fewer packages may be analysed? typically when the user specifies this package number n, exactly n many eligible packages will be analysed

so we look for more eligible packages to meet the user given number

Reply:
yeah, it does...

Maybe just `N` is classic, or `<number-of-packages>` or `<package-count>` or `<package-limit>`

Reply:
Or maybe we should make this an optional option... `--max-packages=1000`, since it's not strictly necessary.

Reply:
yeah I think `<N>` is good

do you mean _another_ option max-packages, so that we don't keep looking for eligible packages past this number?